### DISCUSSION TITLE
tp.date.now returns 1/1/2044

### DISCUSSION THREAD

**Discussion by joshuawagner** [SilentVoid13/Templater#683] [Help] [answered]
Hi friends, can anyone tell me how to resolve an issue with my daily note template? 

# My template is here and my problem is below

```
<%*
	function log(msg) {
		console.log(msg)
	}
	let title = tp.file.title
	let titleBase = "log "
	let titleDate = tp.date.now("YYYY-MM-DD")
	if (title.startsWith("Untitled")) 
	{
		titleDate = await tp.system.prompt("Date", tp.date.now("YYYY-MM-DD"));
		title = titleBase + titleDate
		await tp.file.rename(`${title}`);
		log("Templater: New file: '" + `${title}` + "' from 'tem-is-logDaily✅'");
	}
	if (title.startsWith("log"))
		{
		titleDate = tp.date.now("YYYY-MM-DD",0,tp.date.now("YYYY-MM-DD", 0, tp.file.title, "YYYY-MM-DD"),"YYYY-MM-DD");
		title = titleBase + titleDate
		log("Templater: New file: '" + `${title}` + "' from 'tem-is-logDaily✅'");
	}
%>---
created: [[<% tp.date.now("YYYY-MM-DD",0,tp.date.now("YYYY-MM-DD", 0, tp.file.title, "YYYY-MM-DD"),"YYYY-MM-DD") %>]]
metatableskip: false
characteristics: [[is-logDaily]]
aliases: 
acceptableEmoji: ✅ ⭐ 📚 📖 ✏️ ✨ ☕ ⛳️ ✌️
---
# [[<% tp.date.now("YYYY-MM-DD",0,`${titleDate}`,"YYYY-MM-DD") %> | <% tp.date.now("ddd, MMM Do, YYYY",0,`${titleDate}`,"YYYY-MM-DD") %>]]
[[log/log <% tp.date.now("YYYY-MM-DD",-1,`${titleDate}`,"YYYY-MM-DD") %>|log <% tp.date.now("YYYY-MM-DD",-1,`${titleDate}`,"YYYY-MM-DD") %>]] | [[log/log <% tp.date.now("YYYY-MM-DD",+1,`${titleDate}`,"YYYY-MM-DD") %>|log <% tp.date.now("YYYY-MM-DD",+1,`${titleDate}`,"YYYY-MM-DD") %>]]

# Log
###### 0000
## <% tp.file.cursor(1) %>
# Supplementary
## Process
## Footnotes
```
![20220704_144205-screenshot](https://user-images.githubusercontent.com/1421213/177210275-daa0c1e8-6213-4294-858e-ab783064100d.png)

# Expected Behavior - Scenario 1
Everything resolves beautifully if the note is created by clicking a link for today's uncreated note (`log 2022-07-04 | Mon, Jul 4th, 2022`) in `log\log 2022-07-03.md`.  This causes Templater to use `tem-log-is-daily.md` as expected, as it is named in my Templater settings as the template for my `log` subfolder.

![20220704_145157-screenshot](https://user-images.githubusercontent.com/1421213/177210789-8097bf0e-05a3-4a21-adf8-f43fa024355b.png)

# Problem - Scenario 2
When I choose `Templater: Create new note from Template` it asks for a date as expected.

![20220704_145329-screenshot](https://user-images.githubusercontent.com/1421213/177211104-b759115a-dcd7-4104-8e23-6e08117007a1.png)

![20220704_145458-screenshot](https://user-images.githubusercontent.com/1421213/177211113-2e1e1db6-9a4d-429e-b134-875a0d2593aa.png)

## #PROBLEM 
Today's date is correctly used to name the file but 1/1/2044 is used for the `tp.date.now` function. I thought I had it scripted correctly to use the file name as a reference and that appears to be working correctly when creating the template with the name pre-determined by the link, as in the first scenario.

![20220704_145531-screenshot](https://user-images.githubusercontent.com/1421213/177211893-e5e6f96e-27f4-41f4-88d3-eb51a5530c3a.png)

## #QUESTION
_What do I need to do to cause the script to use the user-input rather than the yet-unnamed file?_

Thanks for your help.

**Comment by AB1908** [comment]
Your `tp.date.now` calls look fairly messy and repetitive. Can you try throwing them in a variable? You also only really need the first argument. The rest are optional.

**Reply by joshuawagner** [reply]
Thanks for the suggestion. I've read a bit on StackOverflow and in this forum and I can see that I can shorten it using `let now = tp.date.now` and use `now` instead of `tp.date.now` . Is that what you mean or is there more that I can do?

**Reply by AB1908** [reply]
Pretty much. Looks like your new template is spot on. Typically, you can get away with ignoring optional arguments unless you actually need them. Optional arguments are those that have a `?` in the docs, for example `reference_format?`.

**Comment by welpdx** [comment, marked as answer]
### two if loops

Are you sure you want two if statements there after another?  Because the first `if` loop changes the file to `log 2022-07-04`, and since the title now begins with `log`, it now goes into the second `if` loop. (On my test, the second if loop now takes the `log 2022-07-04` and turns it into `log 0001-01-01`. 

I think what you want is an if else if. This way, if the first if loop is run, it skips the second...

```js
if (title.startsWith("Untitled")) 
{
titleDate = await tp.system.prompt("Date", tp.date.now("YYYY-MM-DD"));
title = titleBase + titleDate
await tp.file.rename(`${title}`);
log("Templater: New file: '" + `${title}` + "' from 'tem-is-logDaily✅'");
} else if (title.startsWith("log"))
{
titleDate = tp.date.now("YYYY-MM-DD",0,tp.date.now("YYYY-MM-DD", 0, tp.file.title, "YYYY-MM-DD"),"YYYY-MM-DD");
title = titleBase + titleDate
log("Templater: New file: '" + `${title}` + "' from 'tem-is-logDaily✅'");
}
```

### Also, 
> created: [[<% tp.date.now("YYYY-MM-DD",0,tp.date.now("YYYY-MM-DD", 0, tp.file.title, "YYYY-MM-DD"),"YYYY-MM-DD") %>]]

I think it should be: ``[[<% tp.date.now("YYYY-MM-DD",0,tp.date.now("YYYY-MM-DD", 0, `${titleDate}`, "YYYY-MM-DD"),"YYYY-MM-DD") %>]]``, not `tp.file.title` in there right? Since you are not offsetting or anything and simply just outputting the date you entered in, you can make it simpler to just: 
```md
created: [[<% tp.date.now("YYYY-MM-DD",0, `${titleDate}`,"YYYY-MM-DD") %>]]
```

#### Edit:
### Also also
You don't have to do `${titleDate}`. You can call variables just by their names `titleDate`. `${titleDate} == titleDate`

Hope that helps.

~Welpdx

**Reply by joshuawagner** [reply]
Thank you so much! That works beautifully. Yes, my nested ifs were the problem and `${titleDate}` is what I needed.

And you were also correct about the simplification of my `created:` metadata except that I was wrong for wanting an adjusted date there at all. I realize now that I just want `tp.date.now` there.

For reference, here's what I ended up with:

```
<%*
	function log(msg) {
		console.log(msg)
	}
	let title = tp.file.title
	let titleBase = "log "
	let titleDate = tp.date.now("YYYY-MM-DD")
	if (title.startsWith("Untitled")) 
	{
		titleDate = await tp.system.prompt("Date", tp.date.now("YYYY-MM-DD"));
		title = titleBase + titleDate
		await tp.file.rename(`${title}`);
		log("Templater: New file: '" + `${title}` + "' from 'tem-is-logDaily✅'");
	} else if (title.startsWith("log"))
	{
		titleDate = tp.date.now("YYYY-MM-DD",0,tp.date.now("YYYY-MM-DD", 0, tp.file.title, "YYYY-MM-DD"),"YYYY-MM-DD");
		title = titleBase + titleDate
		log("Templater: New file: '" + `${title}` + "' from 'tem-is-logDaily✅'");
	}
%>---
created: [[<% tp.date.now("YYYY-MM-DD") %>]]
metatableskip: false
characteristics: [[is-logDaily]]
aliases: 
acceptableEmoji: ✅ ⭐ 📚 📖 ✏️ ✨ ☕ ⛳️ ✌️
---
# [[<% tp.date.now("YYYY-MM-DD",0,`${titleDate}`,"YYYY-MM-DD") %> | <% tp.date.now("ddd, MMM Do, YYYY",0,`${titleDate}`,"YYYY-MM-DD") %>]]
[[log/log <% tp.date.now("YYYY-MM-DD",-1,`${titleDate}`,"YYYY-MM-DD") %>|log <% tp.date.now("YYYY-MM-DD",-1,`${titleDate}`,"YYYY-MM-DD") %>]] | [[log/log <% tp.date.now("YYYY-MM-DD",+1,`${titleDate}`,"YYYY-MM-DD") %>|log <% tp.date.now("YYYY-MM-DD",+1,`${titleDate}`,"YYYY-MM-DD") %>]]

# Log
###### 0000
## <% tp.file.cursor(1) %>
# Supplementary
## Process
## Footnotes

```

**Reply by welpdx** [reply]
nice!

**Comment by joshuawagner** [comment]
thank you!
